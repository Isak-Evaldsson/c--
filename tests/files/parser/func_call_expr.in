int func() {
    // Check correct parameter list
    int a = func1();
    int b = func2(1);
    int c = func3(a, b, 1);

    // Check correct order in tree
    int d = 5 * func(1 + 1) + 1;
}