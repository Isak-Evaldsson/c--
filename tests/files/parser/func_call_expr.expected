Prototype: func
    VarDecl: a
        FunCallExpr: func1
    VarDecl: b
        FunCallExpr: func2
            LiteralExpr: 1
    VarDecl: c
        FunCallExpr: func3
            VarUseExpr: a
            VarUseExpr: b
            LiteralExpr: 1
    VarDecl: d
        BinOpExpr: +
            BinOpExpr: *
                LiteralExpr: 5
                FunCallExpr: func
                    BinOpExpr: +
                        LiteralExpr: 1
                        LiteralExpr: 1
            LiteralExpr: 1
